<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="6,10">
    <item>
     <layout class="QVBoxLayout" name="main" stretch="1,0,0">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPlainTextEdit" name="edit">
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::NoWrap</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="buttonPanel">
        <item>
         <widget class="QPushButton" name="randomMatrixButton">
          <property name="toolTip">
           <string>Create random matrix</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/svg/matrix-icon.svg</normaloff>:/svg/matrix-icon.svg</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="refreshButton">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/svg/refresh-icon.svg</normaloff>:/svg/refresh-icon.svg</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="optionPanel">
        <item>
         <layout class="QHBoxLayout" name="left">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="iterationsLabel">
              <property name="text">
               <string>Iterations</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="scaleLabel">
              <property name="text">
               <string>Scale</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="coolingFactorLabel">
              <property name="text">
               <string>Cooling Factor</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="springConstantLabel">
              <property name="text">
               <string>Spring constant</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="repulsiveConstantLabel">
              <property name="text">
               <string>Repulsive constant</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="IntLineEdit" name="iterationsEdit">
              <property name="text">
               <string>100</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="DoubleLineEdit" name="scaleEdit">
              <property name="text">
               <string>1.0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="DoubleLineEdit" name="coolingFactorEdit">
              <property name="text">
               <string>0.9</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="DoubleLineEdit" name="springConstantEdit">
              <property name="text">
               <string>0.1</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="DoubleLineEdit" name="repulsiveConstantEdit">
              <property name="text">
               <string>0.1</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="right">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QLabel" name="circleSizeLabel">
                <property name="text">
                 <string>Circle size</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="lineWidthLabel">
                <property name="text">
                 <string>Line width</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="labelFontSizeLabel">
                <property name="text">
                 <string>Label font size</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="IntLineEdit" name="circleSizeEdit">
                <property name="text">
                 <string>4</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="IntLineEdit" name="lineWidthEdit">
                <property name="text">
                 <string>1</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="IntLineEdit" name="labelFontSizeEdit">
                <property name="text">
                 <string>8</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="CanvasView" name="view">
      <property name="styleSheet">
       <string notr="true">background-color: white; border: 1px solid #7A7A7A;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CanvasView</class>
   <extends>QFrame</extends>
   <header>canvasview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>IntLineEdit</class>
   <extends>QLineEdit</extends>
   <header>input/intlineedit.h</header>
  </customwidget>
  <customwidget>
   <class>DoubleLineEdit</class>
   <extends>QLineEdit</extends>
   <header>input/doublelineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
